### PR TITLE
Update to the Lua 5.3 language specification

### PR DESCRIPTION
**Pull request by glguy**
Lua 5.3 adds
Binary operators: << >> | & ~
Unary operators: ~
Unicode literals

### CODE REVIEW

**Comment by osa1** [comment]
Awesome @glguy, thanks for doing this!

I'll have a look at .hs files. What are differences in Lua files? You mentioned some unicode strings etc. could you list your changes so that we at least know what to put back after fixing issues like unicode string lexing etc.?

**Comment by glguy** [comment]
My change was that I didn't include the utf8.lua test file and I saved all the test files in utf-8

File: src/Language/Lua/Annotated/Parser.hs [hunk 1]
```diff
@@ -244,6 +251,11 @@ binop = do
     , tok LTokNotequal >> return (Binop pos (NEQ pos), 3, 3)
     , tok LTokAnd >> return (Binop pos (And pos), 2, 2)
     , tok LTokOr >> return (Binop pos (Or pos), 1, 1)
+    , tok LTokAmpersand >> return (Binop pos (BAnd pos), 1, 1)
+    , tok LTokPipe >> return (Binop pos (BOr pos), 1, 1)
+    , tok LTokTilde >> return (Binop pos (BXor pos), 1, 1)
+    , tok LTokDGT >> return (Binop pos (ShiftR pos), 1, 1)
+    , tok LTokDLT >> return (Binop pos (ShiftL pos), 1, 1)
```
Comment (osa1) [inline src/Language/Lua/Annotated/Parser.hs]: Priorities of these operators are wrong, they should be same as their priorities set in pretty printer below.

**Comment by glguy** [comment]
Oversight for precedences fixed

**Pull request merged**